pkg/plugins/framework/helpers.go

```diff
@@ -0,0 +1,174 @@
+package framework
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/lima-vm/lima/pkg/limayaml"
+)
+
+// Config represents the VM configuration
+type Config struct {
+	limayaml.LimaYAML
+}
+
+// ParseConfig parses the serialized config into a Config struct
+func ParseConfig(serializedConfig string) (*Config, error) {
+	var config Config
+	if err := json.Unmarshal([]byte(serializedConfig), &config); err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+	return &config, nil
+}
+
+// CreateResponse creates a success response
+func CreateResponse(success bool, message string) *plugins.Response {
+	return &plugins.Response{
+		Success: success,
+		Message: message,
+	}
+}
+
+// CreateErrorResponse creates an error response
+func CreateErrorResponse(err error) *plugins.Response {
+	return &plugins.Response{
+		Success: false,
+		Message: err.Error(),
+	}
+}
+
+// EnsureDir ensures a directory exists, creating it if necessary
+func EnsureDir(path string) error {
+	if err := os.MkdirAll(path, 0755); err != nil {
+		return fmt.Errorf("failed to create directory %s: %w", path, err)
+	}
+	return nil
+}
+
+// GetInstanceDir returns the instance directory path
+func GetInstanceDir(instanceID string) string {
+	return filepath.Join(os.Getenv("HOME"), ".lima", instanceID)
+}
+
+// GetInstanceConfigPath returns the instance config file path
+func GetInstanceConfigPath(instanceID string) string {
+	return filepath.Join(GetInstanceDir(instanceID), "lima.yaml")
+}
+
+// GetInstanceDiskPath returns the instance disk file path
+func GetInstanceDiskPath(instanceID string) string {
+	return filepath.Join(GetInstanceDir(instanceID), "disk.img")
+}
+
+// GetInstanceSocketPath returns the instance socket file path
+func GetInstanceSocketPath(instanceID string) string {
+	return filepath.Join(GetInstanceDir(instanceID), "socket")
+}
+
+// GetPluginSocketPath returns the plugin socket file path
+func GetPluginSocketPath(pluginName string) string {
+	return filepath.Join("/tmp", fmt.Sprintf("lima-plugin-%s.sock", pluginName))
```

Comment:
This doesn't allow running multiple Lima instances especially with multiple user accounts on the host.
This is also vulnerable to a race condition.

You can just use stdio to eliminate the sockets
https://github.com/moby/buildkit/blob/de56a3c5056341667b5bad71f414ece70b50724f/frontend/gateway/grpcclient/client.go#L1240